torch/_inductor/utils.py

```diff
@@ -235,6 +235,38 @@ def next_power_of_2(n: int) -> int:
     return n
 
 
+def _type_of(key):
```

Comment:
Please add a comment explaining where this function came from.  Is it copied from Triton?  If so add a link.

Reply:
I have added a reference link. Please help me take a look. Thank you